lib/galaxy/webapps/galaxy/buildapp.py

```diff
@@ -628,6 +628,18 @@ def populate_api_routes(webapp, app):
                           action='delete_custom_builds',
                           conditions=dict(method=["DELETE"]))
 
+    webapp.mapper.connect('set_favorite_tool',
+                          '/api/users/{id}/favorites/tools',
+                          controller='users',
+                          action='set_favorite_tool',
+                          conditions=dict(method=["PUT"]))
+
+    webapp.mapper.connect('remove_favorite_tool',
+                          '/api/users/{id}/favorites/tools',
+                          controller='users',
+                          action='remove_favorite_tool',
+                          conditions=dict(method=["POST"]))
```

Comment:
DELETE should work after https://github.com/galaxyproject/galaxy/pull/6467 is merged, since it upgrades to recent WebOb, which introduced DELETE with a body in version 1.7 https://docs.pylonsproject.org/projects/webob/en/stable/changes.html. But wouldn't it be better to have the route be `/api/users/{id}/favorites/tools/{tool_id}` ? Than POST and DELETE would feel pretty natural to me.

Reply:
>trouble with Routes parsing the wildcard patterns when given the tool_id encoded in the url

Reply:
tried few more things and it seems `/{object_id:.*?}` works, so I will refactor this as well

Reply:
This is now in.